File: src/main/buttons/iconbuttons/Save.java
package main.buttons.iconbuttons;

import main.rooms.Game;
import main.rooms.SaveRoom;

public class Save extends IconButton {

    public Save() {
        super("resources/images/save_button.png");
    }

    @Override
    public void clicked() {
        if (main.getRoom() instanceof Game) {
            if (main.game.getSaveFile() == null) {
                main.switchRoom(main.saveRoom);
            } else {
                main.game.saveGame();
            }
        } else if (main.getRoom() instanceof SaveRoom) {
            main.game.saveGame();
            main.switchRoom(main.game);
        }
    }

}

File: src/main/buttons/iconbuttons/Save.kt
package main.buttons.iconbuttons

import main.Main
import main.rooms.Game
import main.rooms.SaveRoom

class Save : IconButton("resources/images/save_button.png") {
    override fun clicked() {
        if (Main.room is Game) {
            if (Game.saveFile == null) {
                Main.switchRoom(SaveRoom)
            } else {
                Game.saveGame()
            }
        } else if (Main.room is SaveRoom) {
            Game.saveGame()
            Main.switchRoom(Game)
        }
    }
}